### PR TITLE
Status branchonly dax

### PR DESCRIPTION
**Pull request by daxgames**
### Changes

- Syncs your change with Cmder `master` branch.
- Fixes `cmd::Cmder` status false branch only display.
- Set unknown color to white.

image

### CODE REVIEW

**Comment by daxgames** [comment]
I think the `unknown` color should not be 'green' or 'grey' in order to show a difference between the prompt folder and the prompt end 'lambda' character.  I don't know what it should be though so I made it 'white'.  This decision bothers me a little since 'white' normally denotes clean status but I like it better than 'grey'.

**Comment by daxgames** [comment]
My mistake `bash` is working I just had co copy the new `git-prompt.sh` to `vendor\git-for-windows\etc\profile.d\git-prompt.sh`

File: vendor/cmder_prompt_config.lua.default [hunk 1]
```diff
@@ -43,4 +43,4 @@ lamb_color = "\x1b[1;30;40m" -- Light Grey = Lambda Color
 clean_color = "\x1b[1;37;40m"
 dirty_color = "\x1b[33;3m"
 conflict_color = "\x1b[31;1m"
-unknown_color = "\x1b[30;1m"
+unknown_color = "\x1b[37;1" -- White = No VCS Status Branch Color
```
Comment (chrisant996) [inline vendor/cmder_prompt_config.lua.default]: This is missing the `m` at the end of the escape code.

File: vendor/clink.lua [hunk 2]
```diff
@@ -397,35 +397,43 @@ local function git_prompt_filter()
...
+              if gitConflict then
+                  color = colors.conflict
+              end
+
+              clink.prompt.value = string.gsub(clink.prompt.value, "{git}", color.."("..verbatim(branch)..")")
+              return false
+          end
+      end
+    else
+      if git_dir then
```
Comment (chrisant996) [inline vendor/clink.lua]: It looks possible to make a fix without duplicating code:
- Suppose the `color = colors.nostatus` line (432) is moved immediately after the `if branch then` line.
- Suppose the `if cmderGitStatusOptIn then` line is moved immediately after that (i.e. immediately after the new location of the `color = colors.nostatus` line).
- Suppose the matching `end` from the `if cmderGitStatusOptIn then` block is moved immediately before the `clink.prompt.value = ...` line.

I think that will allow the `else ...` block on line 428-436 to be removed, while still fixing the bug.

Thoughts?

**Comment by daxgames** [comment]
@chrisant996 probably I will look.  Was boted this am while the family was asleep, then thay got up so had to stop

**Review by chrisant996 — APPROVED** [review]

**Review by ian-craig — APPROVED** [review]

**Pull request merged**